pkg/backends/calico/client.go

```diff
@@ -202,8 +204,6 @@ func NewCalicoClient(confdConfig *config.Config) (*client, error) {
 			c.OnInSync(SourceRouteGenerator)
```

Comment:
1. This is the error case.  We've failed to create the RG, so it won't ever call OnInSync normally; hence the code here calls OnInSync to balance for that.

2. I'm not sure how far I should let this task expand to fixing all the concurrency problems in confd.  (But time-permitting, I agree it would be better to; and we are in the area now.)

Reply:
I've addressed the readiness calculation concern with https://github.com/projectcalico/confd/pull/323/commits/88abd8141357e6653f519d73b7d08747c3ff3321